doc/nrf/migration/migration_guide_1.x_to_2.x.rst

```diff
@@ -0,0 +1,430 @@
+
+The other important issue is decoding/encoding context access, which is described in :ref:`migration_tinycbor_to_zcbor_mcumgr_ctxt`.
+
+.. note::
+
+   The encoding context provided by the command processing handler function as a parameter has the top level map already created.
+   The map will be closed on a successful return from the handler, so handlers need not create the top level map or close it.
+   The decoding context, in contrary to the encoding, is set before the top map and handlers are responsible for opening of the top-level map as a part of command decoding/processing code.
+
+.. _migration_tinycbor_to_zcbor_mcumgr_ctxt:
+
+Decoding and encoding context
+-----------------------------
+
+Command processing functions, whenever TinyCBOR or zcbor is used, will be given a pointer to an object of type ``struct mgmt_ctxt``.
+TinyCBOR uses the ``encoder`` element of this object for encoding functions and ``it``, a TinyCBOR ``CborValue`` type iterator, for decoding functions.
+
+zcbor uses ``zcbor_state_t`` type objects for decoding and encoding states and these state objects have replaced the TinyCBOR ``it`` iterator and ``encoder``, respectively, within ``struct mgmt_ctxt`` indirectly, as they are embedded within the ``struct cbor_nb_reader`` type object for decoding context, and in the ``struct cbor_nb_writer`` type object for encoding context.
+
+The ``struct cbor_nb_reader`` and ``struct cbor_nb_writer`` objects tie ``zcbor_state_t`` with ``net_buf`` type buffers that hold data for decoding or will hold encoded data.
+
+The replacement of decoder and encoder objects, as accessed through ``struct mgmt_ctxt``, is shown by the following table:
+
+.. table::
+    :align: center
+
+    +===================+===================+
+    | TinyCBOR          | zcbor             |
+    +-------------------+-------------------+
+    | ``encoder``       | ``cnbe.ts``       |
+    |                   | ``cnbe->ts``      |
+    +-------------------+-------------------+
+    | ``it``            | ``cnde.ts``       |
+    |                   | ``cnde->ts``      |
+    +---------------------------------------+
+
+.. _migration_tinycbor_to_zcbor_mcumgr_cborattr:
+
+Replacing `cbor_read_object` with `zcbor_map_decode_bulk`
+---------------------------------------------------------
+
+.. note::
+
+   In scenarios where decoding of keys is not required or it is required for only a single key, the procedure described in this section can be greatly simplified as a single key can be obtained by using ``zcbor_tst_decode`` looped until the key is found.
+
+``cborattr`` was a private MCUMgr utility that was used within the command processing code to process CBOR list contents in bulk and is now replaced with ``zcbor_bulk``.
+These utility APIs are represented by a single function: ``cbor_read_object`` in case of TinyCBOR, and ``zcbor_map_decode_bulk`` in case of zcbor.
```

Comment:
```suggestion
These utility APIs are represented by a single function, ``cbor_read_object()`` in case of TinyCBOR, and ``zcbor_map_decode_bulk()`` in case of zcbor.
```